crates/pubsub/src/implementors/mosquitto.rs

```diff
@@ -0,0 +1,157 @@
+use std::sync::{Arc, Mutex};
+
+use anyhow::{Context, Result};
+use async_channel::Receiver;
+use futures::executor::block_on;
+use mosquitto_rs::{Client, Message, QoS};
+use slight_common::BasicState;
+
+#[derive(Clone)]
+pub struct MosquittoImplementor {
+    mqtt: Arc<Mutex<Client>>,
+    host: String,
+    port: i32,
+    subscriptions: Arc<Mutex<Vec<String>>>,
+    subscriber: Arc<Mutex<Option<Receiver<Message>>>>,
+}
+
+impl std::fmt::Debug for MosquittoImplementor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "MosquittoImplementor")
+    }
+}
+
+// Pub+Sub
+impl MosquittoImplementor {
+    pub fn new(slight_state: &BasicState) -> Self {
+        let mqtt = Client::with_auto_id().unwrap();
+        let host = String::from_utf8(
+            slight_runtime_configs::get(
+                &slight_state.secret_store,
+                "MOSQUITTO_HOST",
+                &slight_state.slightfile_path,
+            )
+            .with_context(|| {
+                format!(
+                    "failed to get 'MOSQUITTO_HOST' secret using secret store type: {}",
+                    slight_state.secret_store
+                )
+            })
+            .unwrap(),
+        )
+        .unwrap();
+
+        let port = String::from_utf8(
+            slight_runtime_configs::get(
+                &slight_state.secret_store,
+                "MOSQUITTO_PORT",
+                &slight_state.slightfile_path,
+            )
+            .with_context(|| {
+                format!(
+                    "failed to get 'MOSQUITTO_PORT' secret using secret store type: {}",
+                    slight_state.secret_store
+                )
+            })
+            .unwrap(),
+        )
+        .unwrap()
```

Comment:
There is a lot of `unwrap()`. My last comment about making this function returning a `anyhow::Result<>` could remove some of the unwraps.

Reply:
If any of these things fail, I want to panic immediately and want to get the unwrap message.